### PR TITLE
Feature: Input::Checkbox

### PR DESCRIPTION
**Pull request by richardverheyen**
This PR has the new pods/components/input/checkbox component
As of yet it seems that we don't asynchronously save attributes on our models, but I did build a 'saving' state which can be passed and overrides the 'value = true' state. 
```
if (saving) {
 showEllipsis()
} (else if value) {
 showCheck()
}
```
The saving state uses background: repeating-linear-background and a keyframe animation which are both fully supported in ie10 and above. 

NOTE:
I did force the 'saving' state in the product.featured input, which we might not want for this PR, also it's not extended to other model modals.

### CODE REVIEW

File: app/pods/secure/products/product/template.hbs [hunk 1]
```diff
@@ -38,69 +38,48 @@
...
-        @text="Public?"
-        @hint="If yes, this product will be shown on app.interflux.com. If no, this product is hidden from the world and can only be seen by Interflux members who are logged in on admin.interflux.com."
-      /> -->
-      <Input::Boolean
+      <Input::Checkbox
         @value={{product.public}}
-        @textTrue="Public"
-        @textFalse="Private"
+        @textLabel="Visible on website"
         @onToggle={{fn (mut product.public)}}
```
Comment (janwerkhoven) [inline app/pods/secure/products/product/template.hbs:49]: ❤️

File: app/pods/secure/products/product/template.hbs [hunk 2]
```diff
@@ -38,69 +38,48 @@
...
     <Fieldset>
-      <!-- <Label
-        @text="Featured?"
-      /> -->
-      <Input::Boolean
-        @value={{product.featured}}
-        @textTrue="Featured"
-        @textFalse="Normal"
-        @onToggle={{fn (mut product.featured)}}
+      <Input::Checkbox
```
Comment (janwerkhoven) [inline app/pods/secure/products/product/template.hbs:54]: I'm on the fence whether to name these components after the UI (checkbox, toggle, ...) or after the data type (boolean, string, one-to-many). Booleans could be set with toggles, checkboxes and buttons alike. In this case `Input::Boolean` is already taken by a toggle UI, so you did the right thing namespacing to `Input::Checkbox`.

File: app/pods/secure/products/product/template.hbs [hunk 3]
```diff
@@ -38,69 +38,48 @@
...
-      /> -->
-      <Input::Boolean
-        @value={{product.new}}
-        @textTrue="Mark as new"
-        @textFalse="Normal"
-        @onToggle={{fn (mut product.new)}}
+      <Input::Checkbox
+        @value={{product.featured}}
+        @textLabel='Mark as "featured" on website'
+        @saving=true
```
Comment (janwerkhoven) [inline app/pods/secure/products/product/template.hbs:73]: Good, this keeps the component a dumb view-only component at the end of the component hierarchy. The smarter components above will know about saving.

File: app/pods/components/input/checkbox/template.hbs [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<div class="input checkbox {{@value}} {{if @saving 'saving'}}" {{on 'click' this.toggle}}>
+  <div>
+    {{#if @saving}}
+      <Svg::Ellipsis @fill="#fff" />
+    {{else if @value}}
+      <Svg::Check @fill="#fff" />
```
Comment (janwerkhoven) [inline app/pods/components/input/checkbox/template.hbs:6]: Good approach. This is what I started out with, but then reverted from it.
1. The template doesn't care about styling, CSS files do, separation of concerns.
2. By fixating the colour in the HTML, we aren't able to leverage CSS `:hover` and `@media` to set different colours.

**Review by janwerkhoven — APPROVED** [review]
Looking good! I'd request only one change, which is to use CSS to set `svg [fill]` colours instead of HTML. Not a major sticking point if we don't. Merge at will. ⛵

File: app/pods/secure/products/product/template.hbs (hunk 2 above)
Comment (richardverheyen) [reply app/pods/secure/products/product/template.hbs:54]: It was my assumption that similar to Salesforce's ootb components there would be a 'CheckboxGroup' or similar to support groups of 'real' checkbox inputs

File: app/pods/components/input/checkbox/template.hbs (hunk 4 above)
Comment (richardverheyen) [reply app/pods/components/input/checkbox/template.hbs]: I understand, makes sense!

**Pull request merged**